zcash_primitives/src/transaction/sighash_v5.rs

```diff
@@ -170,22 +179,20 @@ pub fn v5_signature_hash<
     signable_input: &SignableInput<'_>,
     txid_parts: &TxDigests<Blake2bHash>,
 ) -> Blake2bHash {
+    // The caller must always provide the transparent digests if the transaction has a
+    // transparent component.
+    assert!(tx.transparent_bundle.is_some() ^ txid_parts.transparent_digests.is_none());
```

Comment:
Why use `^` instead of `||` here?

Reply:
I want to ensure both cases: either both are `Some` or both are `None`. What I need here is an X OR, which I emulate by swapping one of the inputs around.

Reply:
```suggestion
    assert!(tx.transparent_bundle.is_some() == txid_parts.transparent_digests.is_some());
```
https://www.cs.utexas.edu/users/EWD/transcriptions/EWD10xx/EWD1070.html :-)